addons/binding/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/discovery/HDPowerViewSceneDiscoveryService.java

```diff
@@ -33,39 +34,59 @@
 
     private final Logger logger = LoggerFactory.getLogger(HDPowerViewSceneDiscoveryService.class);
     private final HDPowerViewHubHandler hub;
+    private final Runnable scanner;
+    private ScheduledFuture<?> backgroundFuture;
 
     public HDPowerViewSceneDiscoveryService(HDPowerViewHubHandler hub) {
-        super(Collections.singleton(HDPowerViewBindingConstants.THING_TYPE_SCENE), 600, true);
+        super(Collections.emptySet(), 600, true);
         this.hub = hub;
+
+        scanner = createScanner();
     }
 
     @Override
     protected void startScan() {
-        HDPowerViewWebTargets targets = hub.getWebTargets();
-        Scenes scenes;
-        try {
-            scenes = targets.getScenes();
-        } catch (IOException e) {
-            logger.error(e.getMessage(), e);
-            stopScan();
-            return;
-        }
-        if (scenes != null) {
-            for (Scene scene : scenes.sceneData) {
-                ThingUID thingUID = new ThingUID(HDPowerViewBindingConstants.THING_TYPE_SCENE,
-                        Integer.toString(scene.id));
-                DiscoveryResult result = DiscoveryResultBuilder.create(thingUID)
-                        .withProperty(HDPowerViewSceneConfiguration.ID, scene.id).withLabel(scene.getName())
-                        .withBridge(hub.getThing().getUID()).build();
-                thingDiscovered(result);
-            }
-        }
-        stopScan();
+        scheduler.execute(scanner);
     }
 
     @Override
     protected void startBackgroundDiscovery() {
-        startScan();
+        if (backgroundFuture != null && !backgroundFuture.isDone()) {
+            backgroundFuture.cancel(true);
+            backgroundFuture = null;
+        }
+        backgroundFuture = scheduler.scheduleAtFixedRate(scanner, 0, 60, TimeUnit.SECONDS);
```

Comment:
scheduleWithFixedDelay is usually better, since it avoids jobs piling up (e.g. while being in a breakpoint while debugging).